### PR TITLE
[ALLUXIO-2689] New journal to improve master's availability [PR3]

### PR DESCRIPTION
**Pull request by peisun1115**
PR3:
The rest of the changes. It includes the master changes to use the new journal.

Full PR: #5019

### CODE REVIEW

File: bin/alluxio-stop.sh [hunk 1]
```diff
@@ -45,30 +50,43 @@ stop_proxies() {
...
 
+
 WHAT=${1:--h}
 
 case "${WHAT}" in
   all)
     stop_proxies
     stop_workers
-    stop_proxy
+    stop_proxies
```
Comment (jsimsa) [inline bin/alluxio-stop.sh]: undo

File: core/common/src/main/java/alluxio/PropertyKey.java [hunk 2]
```diff
@@ -125,12 +125,18 @@
   MASTER_JOURNAL_FLUSH_BATCH_TIME_MS(Name.MASTER_JOURNAL_FLUSH_BATCH_TIME_MS, 5),
   MASTER_JOURNAL_FLUSH_TIMEOUT_MS(Name.MASTER_JOURNAL_FLUSH_TIMEOUT_MS, 300000),
   MASTER_JOURNAL_FOLDER(Name.MASTER_JOURNAL_FOLDER, String.format("${%s}/journal", Name.WORK_DIR)),
+  // Deprecated since 1.5.0 and will be removed in 2.0.
   MASTER_JOURNAL_FORMATTER_CLASS(Name.MASTER_JOURNAL_FORMATTER_CLASS,
-      "alluxio.master.journal.ProtoBufJournalFormatter"),
+      "alluxio.master.journalv0.ProtoBufJournalFormatter"),
```
Comment (jsimsa) [inline core/common/src/main/java/alluxio/PropertyKey.java]: we should not change the property name

File: bin/alluxio-stop.sh (hunk 1 above)
Comment (peisun1115) [inline bin/alluxio-stop.sh]: done

File: core/common/src/main/java/alluxio/PropertyKey.java (hunk 2 above)
Comment (jsimsa) [inline core/common/src/main/java/alluxio/PropertyKey.java]: nevermind, ignore this

File: core/server/common/src/main/java/alluxio/master/journalv0/AsyncJournalWriter.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
+/*
```
Comment (jsimsa) [inline core/server/common/src/main/java/alluxio/master/journalv0/AsyncJournalWriter.java]: optional: I would slightly prefer to use `alluxio.master.journal.v0` as the package

**Pull request closed without merging**